### PR TITLE
Update checkbox logic copy mailing address

### PR DESCRIPTION
**Pull request by sanlouise**
## Description
Behavior of the checkbox (**bold** is new)
- Is unchecked upon first render of the Home Address Edit View
- It is checked if the user has set their Home Address equal to their Mailing Address, either while editing the Mailing address, or in a previous session
- **The user can now explicitly uncheck the box**
- **Unchecking a previously checked box clears out the form** 
- **After clearing the form, the user can still re-check the box to set their Home Address equal to their Mailing Address**

## Testing done
Works well locally.

## Acceptance criteria
- [x] Updated checkbox behavior per ticket's criteria

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/user/profile/vap-svc/containers/CopyMailingAddress.jsx [hunk 1]
```diff
@@ -58,27 +92,13 @@ export function mapStateToProps(state) {
...
-      return false;
-    }
-    return isEqual(
-      pickBy(pick(mailingAddress, ADDRESS_PROPS), e => !!e),
-      pick(residentialAddress, ADDRESS_PROPS),
-    );
-  };
-
   return {
+    emptyMailingAddress: isEmptyAddress(mailingAddress),
```
Comment (erikphansen) [inline src/platform/user/profile/vap-svc/containers/CopyMailingAddress.jsx:96]: Why the rename of this prop?

File: src/platform/user/profile/vap-svc/containers/CopyMailingAddress.jsx [hunk 2]
```diff
@@ -23,26 +24,59 @@ const ADDRESS_PROPS = [
...
+
+    return isEqual(mailingAddressFields, residentialAddressFields);
+  };
+
+  onChange = () => {
+    const { copyMailingAddress, mailingAddress } = this.props;
+
+    // If mailing + home addresses are the same, clear the home address
+    if (this.areHomeMailingAddressesEqual()) {
+      const clearedHomeAddress = mapValues(mailingAddress, () => null);
```
Comment (erikphansen) [inline src/platform/user/profile/vap-svc/containers/CopyMailingAddress.jsx:53]: could call this `emptyAddress` or `blankAddress`

**Review by erikphansen — APPROVED** [review]
Made a few comments that you should feel free to ignore!

File: src/platform/user/profile/vap-svc/containers/CopyMailingAddress.jsx (hunk 1 above)
Comment (sanlouise) [reply src/platform/user/profile/vap-svc/containers/CopyMailingAddress.jsx:96]: No specific reason, I  reorganized this file quite a bit and then made this change somewhere along the way, unaware of this change.

File: src/platform/user/profile/vap-svc/containers/CopyMailingAddress.jsx [hunk 3]
```diff
@@ -23,26 +24,59 @@ const ADDRESS_PROPS = [
...
-    if (event.target.checked) {
-      const address = pick(this.props.mailingAddress, ADDRESS_PROPS);
-      this.props.copyMailingAddress(address);
+  static propTypes = {
+    emptyMailingAddress: PropTypes.bool.isRequired,
+    mailingAddress: PropTypes.object.isRequired,
+    residentialAddress: PropTypes.object.isRequired,
+  };
+
+  areHomeMailingAddressesEqual = () => {
```
[the remark this answers is not in the document]
Comment (sanlouise) [reply src/platform/user/profile/vap-svc/containers/CopyMailingAddress.jsx:33]: I could see that work too 😄 If you're not too opinionated, I'll keep it as is.

**Pull request merged**